pkgs/development/tools/delve/default.nix

```diff
@@ -17,7 +17,20 @@ buildGoModule rec {
 
   nativeBuildInputs = [ makeWrapper ];
 
-  checkFlags = [ "-short" ];
+  preCheck = ''
+    XDG_CONFIG_HOME=$(mktemp -d)
+
+    # The tests use -O0 which is not compatible with -D_FORTIFY_SOURCE.
+    NIX_HARDENING_ENABLE=''${NIX_HARDENING_ENABLE/fortify/}
+  '';
```

Comment:
```suggestion
  hardeningDisable = [ "fortify" ];

  preCheck = ''
    XDG_CONFIG_HOME=$(mktemp -d)
  '';
```

Reply:
In fact I think we can remove `wrapProgram` from postInstall as well.

```
   postInstall = ''
     # fortify source breaks build since delve compiles with -O0
     wrapProgram $out/bin/dlv \
       --prefix disableHardening " " fortify
```

Reply:
> I do not understand how setting `disableHardening=fortify` environment variable in the wrapper achieved disabling fortify checks at runtime.

Oh. I see. This has nothing to do with runtime (as we don't do _FORTIFY_SOURCE=3 anway) but because delve may end up compiling some code that and this environment variable needs to passed to disable Fortify.

I think this should be changed similarly to the change in goland, as discussed in #154016 (/cc @diogox).